### DISCUSSION TITLE
Purpose of `secretWhitelist` parameter in Helm Chart?

### DISCUSSION THREAD

**Discussion by vitalyrychkov** [argoproj/argo-workflows#9816] [Q&A] [answered]
Could not find an answer in the Docu or Google, so, could anyone answer me what is the use of the secretWhitelist parameter?
Could I specify there a secret from a namespace different from Argo workflows namespace?
The description in the Helm values file seems to be related to a specific use case:
```
    ## When present, restricts secrets the server can read to a given list.
    ## You can use it to restrict the server to only be able to access the
    ## service account token secrets that are associated with service accounts
    ## used for authorization.
       secretWhitelist: []
```

**Comment by agilgur5** [comment, marked as answer]
This would actually be a question for [`argo-helm`](https://github.com/argoproj/argo-helm) as this is specific to the Helm chart.

`.Values.secretWhitelist` is used by the [Server's ClusterRole](https://github.com/argoproj/argo-helm/blob/224a227fa49b00fabfb535825c9bd5648f16a30a/charts/argo-workflows/templates/server/server-cluster-roles.yaml#L74) to set [`resourceNames`](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#referring-to-resources).
It is a security measure to decrease the level of access the Server has, as having access to all Secrets makes for a high value attack target. As mentioned in the comments you quoted, the Server only accesses Secrets of ServiceAccounts that are used by users in the UI.

> Could I specify there a secret from a namespace different from Argo workflows namespace?

The Helm chart uses a `ClusterRole` and, if you're not using `.Values.singleNamespace`, a [`ClusterRoleBinding`](https://github.com/argoproj/argo-helm/blob/224a227fa49b00fabfb535825c9bd5648f16a30a/charts/argo-workflows/templates/server/server-crb.yaml#L6), so yes, it can have access to secrets in all namespaces.
